File: parley-network/src/main/java/nu/parleynetwork/android/data/repository/EventRepositoryImpl.java
package nu.parleynetwork.android.data.repository;

import nu.parley.android.data.net.Connectivity;
import nu.parley.android.data.net.service.EventService;
import nu.parley.android.data.repository.EventRepository;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class EventRepositoryImpl implements EventRepository {

    public void fire(String event) {
        Call<Void> eventCall = Connectivity.getRetrofit().create(EventService.class).fire(event);
        eventCall.enqueue(new Callback<Void>() {
            @Override
            public void onResponse(Call<Void> call, Response<Void> response) {
                // Ignore
            }

            @Override
            public void onFailure(Call<Void> call, Throwable t) {
                // Ignore
            }
        });
    }
}

File: parley-network/src/main/java/nu/parleynetwork/android/data/repository/EventRepositoryImpl.kt
package nu.parleynetwork.android.data.repository

import nu.parley.android.data.net.Connectivity
import nu.parley.android.data.net.service.EventService
import nu.parley.android.data.repository.EventRepository
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class EventRepositoryImpl : EventRepository {
    public override fun fire(event: String) {
        var eventCall = Connectivity.getRetrofit().create(
            EventService::class.java
        ).fire(event)
        eventCall.enqueue(object : Callback<Void?> {
            public override fun onResponse(call: Call<Void?>, response: Response<Void?>) {
                // Ignore
            }

            public override fun onFailure(call: Call<Void?>, t: Throwable) {
                // Ignore
            }
        })
    }
}
